wapitiCore/attack/mod_log4shell.py

```diff
@@ -26,54 +28,94 @@ class ModuleLog4Shell(Attack):
 
     def __init__(self, crawler, persister, attack_options, stop_event):
         Attack.__init__(self, crawler, persister, attack_options, stop_event)
-        if not self.is_valid_dns(attack_options.get("dns_endpoint")):
+        if not self._is_valid_dns(attack_options.get("dns_endpoint")):
             self.finished = True
 
     async def must_attack(self, request: Request):
         if self.finished is True:
             return False
         return True
 
-    async def read_headers(self):
+    def read_headers(self):
         with open(path_join(self.DATA_DIR, self.HEADERS_FILE), encoding='utf-8') as headers_file:
             return headers_file.read().split("\n")
 
     async def attack(self, request: Request):
-        headers = await self.read_headers()
+        headers = self.read_headers()
 
-        batch_malicious_headers, headers_uuid_record = await self._get_malicious_headers(headers)
+        batch_malicious_headers, headers_uuid_record = self._get_malicious_headers(headers)
 
         for malicious_headers in batch_malicious_headers:
             modified_request = Request(request.url)
+
             try:
                 await self.crawler.async_send(modified_request, malicious_headers, follow_redirects=True)
             except RequestError:
                 self.network_errors += 1
                 continue
-            await self._verify_headers(modified_request, malicious_headers, headers_uuid_record)
+            await self._verify_headers_vulnerability(modified_request, malicious_headers, headers_uuid_record)
+
+        injected_get_and_post_requests: Iterable[Tuple[Request, str, uuid.UUID]] = itertools.chain(
+            self._inject_payload(request, request.get_params),
+            self._inject_payload(request, request.post_params)
+        )
 
-    async def _verify_headers(self, modified_request: Request, malicious_headers: dict, headers_uuid_record: dict):
+        for malicious_request, param_name, param_uuid in injected_get_and_post_requests:
+            try:
+                await self.crawler.async_send(malicious_request, follow_redirects=True)
+            except RequestError:
+                self.network_errors += 1
+                continue
+            await self._verify_param_vulnerability(malicious_request, param_uuid, param_name)
+
+    async def _verify_param_vulnerability(self, request: Request, param_uuid: uuid.UUID, param_name: str):
+        if not self._verify_dns(str(param_uuid)):
+            return
+        element_type = "query parameter" if request.method == "GET" else "body parameter"
+
+        await self.add_vuln_critical(
+            category=NAME,
+            request=request,
+            info=_("URL {0} seems vulnerable to Log4Shell attack by using the {1} {2}") \
+                .format(request.url, element_type, param_name),
+            parameter=f"{request.get_params if request.method == 'GET' else request.post_params}"
```

Comment:
Why don't you use `param_name` here ?

Reply:
We wanted to show the full payload to the user. If we replace this with the `param_name`
 where can we write it? @bretfourbe

Reply:
If I'm not wrong the payload will appear in the report as the `request` object is dumped using `http_repr`

Reply:
Yes that's true, I'm modifying it